File: sur-base-support/sur-bus/src/main/webapp/assets/standard/standard.js
layui.use(['table', 'admin', 'ax', 'func'], function () {
    var $ = layui.$;
    var table = layui.table;
    var $ax = layui.ax;
    var admin = layui.admin;
    var func = layui.func;

    /**
     * 地层信息管理
     */
    var Standard = {
        tableId: "standardTable"
    };

    /**
     * 初始化表格的列
     */
    Standard.initColumn = function () {
        return [[
            {type: 'checkbox'},
            {field: 'id', hide: true, title: '主键'},
            {field: 'itemId', sort: true, title: '工程id'},
            {field: 'holeCode', sort: true, title: '钻孔编号'},
            {field: 'depth', sort: true, title: '层底深度（米）'},
            {field: 'tcxh', sort: true, title: ''},
            {field: 'mainCode', sort: true, title: '主层编号'},
            {field: 'secondaryCode', sort: true, title: '亚层编号'},
            {field: 'thirdCode', sort: true, title: '次亚层编号'},
            {field: 'tcdzsd', sort: true, title: '地质时代'},
            {field: 'tcdzcy', sort: true, title: '地质成因'},
            {field: 'tcdchd', sort: true, title: '自然层厚度(m)'},
            {field: 'tchd', sort: true, title: '地层厚度(m)'},
            {field: 'type', sort: true, title: '岩土类别'},
            {field: 'name', sort: true, title: '岩土名称'},
            {field: 'tcmdh', sort: true, title: '土名代号'},
            {field: 'tcymc', sort: true, title: '亚岩土名称'},
            {field: 'tcys', sort: true, title: '土层颜色'},
            {field: 'tcmsd', sort: true, title: '密实度'},
            {field: 'tcsid', sort: true, title: '湿度'},
            {field: 'tcksx', sort: true, title: '可塑性'},
            {field: 'tchyd', sort: true, title: '浑圆度'},
            {field: 'tcjyx', sort: true, title: '均匀性'},
            {field: 'tcfhcd', sort: true, title: '风化程度'},
            {field: 'tcysqx', sort: true, title: '岩层倾向(度)'},
            {field: 'tcysqj', sort: true, title: '岩层倾角(度)'},
            {field: 'tckwcf', sort: true, title: '矿物成分'},
            {field: 'tcjggz', sort: true, title: '结构构造'},
            {field: 'tcbhw', sort: true, title: '包含物'},
            {field: 'tcqw', sort: true, title: '气味'},
            {field: 'desc', sort: true, title: '土层描述'},
            {field: 'tcztx', sort: true, title: '完整程度'},
            {field: 'tcjycd', sort: true, title: '坚硬程度'},
            {field: 'tcpl', sort: true, title: '破碎程度'},
            {field: 'tcjlfy', sort: true, title: '节理发育'},
            {field: 'tcdcms', sort: true, title: '地层描述'},
            {field: 'tcjljj', sort: true, title: '节理间距(cm)'},
            {field: 'tcpxjd', sort: true, title: '视倾角(度)'},
            {field: 'tcqttz', sort: true, title: '其他特征'},
            {field: 'tczrzd', sort: true, title: '天然重度(kN/m3)'},
            {field: 'tcnjl', sort: true, title: '粘聚力(kPa)'},
            {field: 'tcnmcj', sort: true, title: '内摩擦角(°)'},
            {field: 'tcczl', sort: true, title: '承载力(kPa)'},
            {field: 'tcysml', sort: true, title: '压缩模量(MPa)'},
            {field: 'tcyzzCzltzz', sort: true, title: '预制桩侧阻力特征值(kPa)'},
            {field: 'tcyzzDzltzz', sort: true, title: '预制桩端阻力特征值(kPa)'},
            {field: 'tcyzzCzlbzz', sort: true, title: '预制桩极限侧阻力标准值(kPa)'},
            {field: 'tcyzzDzlbzz', sort: true, title: '预制桩极限端阻力标准值(kPa)'},
            {field: 'tcckzCzltzz', sort: true, title: '冲(钻)孔桩侧阻力特征值(kPa)'},
            {field: 'tcckzDzltzz', sort: true, title: '冲(钻)孔桩端阻力特征值(kPa)'},
            {field: 'tcckzCzlbzz', sort: true, title: '冲(钻)孔桩极限侧阻力标准值(kPa)'},
            {field: 'tcckzDzlbzz', sort: true, title: '冲(钻)孔桩极限端阻力标准值(kPa)'},
            {field: 'tcuserdefine1', sort: true, title: '自定义字段1'},
            {field: 'tcuserdefine2', sort: true, title: '自定义字段2'},
            {field: 'tcuserdefine3', sort: true, title: '自定义字段3'},
            {align: 'center', toolbar: '#tableBar', title: '操作'}
        ]];
    };

    /**
     * 点击查询按钮
     */
    Standard.search = function () {
        var queryData = {};


        table.reload(Standard.tableId, {
            where: queryData, page: {curr: 1}
        });
    };

    /**
     * 弹出添加对话框
     */
    Standard.openAddDlg = function () {
        func.open({
            title: '添加地层信息',
            content: Feng.ctxPath + '/standard/add',
            tableId: Standard.tableId
        });
    };

     /**
      * 点击编辑
      *
      * @param data 点击按钮时候的行数据
      */
      Standard.openEditDlg = function (data) {
          func.open({
              title: '修改地层信息',
              content: Feng.ctxPath + '/standard/edit?id=' + data.id,
              tableId: Standard.tableId
          });
      };


    /**
     * 导出excel按钮
     */
    Standard.exportExcel = function () {
        var checkRows = table.checkStatus(Standard.tableId);
        if (checkRows.data.length === 0) {
            Feng.error("请选择要导出的数据");
        } else {
            table.exportFile(tableResult.config.id, checkRows.data, 'xls');
        }
    };

    /**
     * 点击删除
     *
     * @param data 点击按钮时候的行数据
     */
    Standard.onDeleteItem = function (data) {
        var operation = function () {
            var ajax = new $ax(Feng.ctxPath + "/standard/delete", function (data) {
                Feng.success("删除成功!");
                table.reload(Standard.tableId);
            }, function (data) {
                Feng.error("删除失败!" + data.responseJSON.message + "!");
            });
            ajax.set("id", data.id);
            ajax.start();
        };
        Feng.confirm("是否删除?", operation);
    };

    // 渲染表格
    var tableResult = table.render({
        elem: '#' + Standard.tableId,
        url: Feng.ctxPath + '/standard/list',
        page: true,
        height: "full-158",
        cellMinWidth: 100,
        cols: Standard.initColumn()
    });

    // 搜索按钮点击事件
    $('#btnSearch').click(function () {
        Standard.search();
    });

    // 添加按钮点击事件
    $('#btnAdd').click(function () {

    Standard.openAddDlg();

    });

    // 导出excel
    $('#btnExp').click(function () {
        Standard.exportExcel();
    });

    // 工具条点击事件
    table.on('tool(' + Standard.tableId + ')', function (obj) {
        var data = obj.data;
        var layEvent = obj.event;

        if (layEvent === 'edit') {
            Standard.openEditDlg(data);
        } else if (layEvent === 'delete') {
            Standard.onDeleteItem(data);
        }
    });
});

File: sur-base-support/sur-bus/src/main/java/com/lfxwkj/sur/entity/Standard.java
package com.lfxwkj.sur.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableField;
import java.io.Serializable;

/**
 * <p>
 * 地层信息
 * </p>
 *
 * @author 张童
 * @since 2020-09-15
 */
@TableName("lz_standard")
public class Standard implements Serializable {

    private static final long serialVersionUID=1L;

    /**
     * 主键
     */
      @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 工程id
     */
    @TableField("item_id")
    private Long itemId;

    /**
     * 钻孔编号
     */
    @TableField("hole_code")
    private String holeCode;

    /**
     * 层底深度（米）
     */
    @TableField("depth")
    private Double depth;

    @TableField("tcxh")
    private Integer tcxh;

    /**
     * 主层编号
     */
    @TableField("main_code")
    private String mainCode;

    /**
     * 亚层编号
     */
    @TableField("secondary_code")
    private String secondaryCode;

    /**
     * 次亚层编号
     */
    @TableField("third_code")
    private String thirdCode;

    /**
     * 地质时代
     */
    @TableField("tcdzsd")
    private String tcdzsd;

    /**
     * 地质成因
     */
    @TableField("tcdzcy")
    private String tcdzcy;

    /**
     * 自然层厚度(m)
     */
    @TableField("tcdchd")
    private Double tcdchd;

    /**
     * 地层厚度(m)
     */
    @TableField("tchd")
    private Double tchd;

    /**
     * 岩土类别
     */
    @TableField("type")
    private String type;

    /**
     * 岩土名称
     */
    @TableField("name")
    private String name;

    /**
     * 土名代号
     */
    @TableField("tcmdh")
    private String tcmdh;

    /**
     * 亚岩土名称
     */
    @TableField("tcymc")
    private String tcymc;

    /**
     * 土层颜色
     */
    @TableField("tcys")
    private String tcys;

    /**
     * 密实度
     */
    @TableField("tcmsd")
    private String tcmsd;

    /**
     * 湿度
     */
    @TableField("tcsid")
    private String tcsid;

    /**
     * 可塑性
     */
    @TableField("tcksx")
    private String tcksx;

    /**
     * 浑圆度
     */
    @TableField("tchyd")
    private String tchyd;

    /**
     * 均匀性
     */
    @TableField("tcjyx")
    private String tcjyx;

    /**
     * 风化程度
     */
    @TableField("tcfhcd")
    private String tcfhcd;

    /**
     * 岩层倾向(度)
     */
    @TableField("tcysqx")
    private Double tcysqx;

    /**
     * 岩层倾角(度)
     */
    @TableField("tcysqj")
    private Double tcysqj;

    /**
     * 矿物成分
     */
    @TableField("tckwcf")
    private String tckwcf;

    /**
     * 结构构造
     */
    @TableField("tcjggz")
    private String tcjggz;

    /**
     * 包含物
     */
    @TableField("tcbhw")
    private String tcbhw;

    /**
     * 气味
     */
    @TableField("tcqw")
    private String tcqw;

    /**
     * 土层描述
     */
    @TableField("tcms")
    private String tcms;

    /**
     * 完整程度
     */
    @TableField("tcztx")
    private String tcztx;

    /**
     * 坚硬程度
     */
    @TableField("tcjycd")
    private String tcjycd;

    /**
     * 破碎程度
     */
    @TableField("tcpl")
    private String tcpl;

    /**
     * 节理发育
     */
    @TableField("tcjlfy")
    private String tcjlfy;

    /**
     * 地层描述
     */
    @TableField("tcdcms")
    private String tcdcms;

    /**
     * 节理间距(cm)
     */
    @TableField("tcjljj")
    private Double tcjljj;

    /**
     * 视倾角(度)
     */
    @TableField("tcpxjd")
    private Double tcpxjd;

    /**
     * 其他特征
     */
    @TableField("tcqttz")
    private String tcqttz;

    /**
     * 天然重度(kN/m3)
     */
    @TableField("tczrzd")
    private Double tczrzd;

    /**
     * 粘聚力(kPa)
     */
    @TableField("tcnjl")
    private Double tcnjl;

    /**
     * 内摩擦角(°)
     */
    @TableField("tcnmcj")
    private Double tcnmcj;

    /**
     * 承载力(kPa)
     */
    @TableField("tcczl")
    private Double tcczl;

    /**
     * 压缩模量(MPa)
     */
    @TableField("tcysml")
    private Double tcysml;

    /**
     * 预制桩侧阻力特征值(kPa)
     */
    @TableField("tcyzz_czltzz")
    private Double tcyzzCzltzz;

    /**
     * 预制桩端阻力特征值(kPa)
     */
    @TableField("tcyzz_dzltzz")
    private Double tcyzzDzltzz;

    /**
     * 预制桩极限侧阻力标准值(kPa)
     */
    @TableField("tcyzz_czlbzz")
    private Double tcyzzCzlbzz;

    /**
     * 预制桩极限端阻力标准值(kPa)
     */
    @TableField("tcyzz_dzlbzz")
    private Double tcyzzDzlbzz;

    /**
     * 冲(钻)孔桩侧阻力特征值(kPa)
     */
    @TableField("tcckz_czltzz")
    private Double tcckzCzltzz;

    /**
     * 冲(钻)孔桩端阻力特征值(kPa)
     */
    @TableField("tcckz_dzltzz")
    private Double tcckzDzltzz;

    /**
     * 冲(钻)孔桩极限侧阻力标准值(kPa)
     */
    @TableField("tcckz_czlbzz")
    private Double tcckzCzlbzz;

    /**
     * 冲(钻)孔桩极限端阻力标准值(kPa)
     */
    @TableField("tcckz_dzlbzz")
    private Double tcckzDzlbzz;

    /**
     * 自定义字段1
     */
    @TableField("tcuserdefine1")
    private String tcuserdefine1;

    /**
     * 自定义字段2
     */
    @TableField("tcuserdefine2")
    private String tcuserdefine2;

    /**
     * 自定义字段3
     */
    @TableField("tcuserdefine3")
    private String tcuserdefine3;


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getItemId() {
        return itemId;
    }

    public void setItemId(Long itemId) {
        this.itemId = itemId;
    }

    public String getHoleCode() {
        return holeCode;
    }

    public void setHoleCode(String holeCode) {
        this.holeCode = holeCode;
    }

    public Double getDepth() {
        return depth;
    }

    public void setDepth(Double depth) {
        this.depth = depth;
    }

    public Integer getTcxh() {
        return tcxh;
    }

    public void setTcxh(Integer tcxh) {
        this.tcxh = tcxh;
    }

    public String getMainCode() {
        return mainCode;
    }

    public void setMainCode(String mainCode) {
        this.mainCode = mainCode;
    }

    public String getSecondaryCode() {
        return secondaryCode;
    }

    public void setSecondaryCode(String secondaryCode) {
        this.secondaryCode = secondaryCode;
    }

    public String getThirdCode() {
        return thirdCode;
    }

    public void setThirdCode(String thirdCode) {
        this.thirdCode = thirdCode;
    }

    public String getTcdzsd() {
        return tcdzsd;
    }

    public void setTcdzsd(String tcdzsd) {
        this.tcdzsd = tcdzsd;
    }

    public String getTcdzcy() {
        return tcdzcy;
    }

    public void setTcdzcy(String tcdzcy) {
        this.tcdzcy = tcdzcy;
    }

    public Double getTcdchd() {
        return tcdchd;
    }

    public void setTcdchd(Double tcdchd) {
        this.tcdchd = tcdchd;
    }

    public Double getTchd() {
        return tchd;
    }

    public void setTchd(Double tchd) {
        this.tchd = tchd;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getTcmdh() {
        return tcmdh;
    }

    public void setTcmdh(String tcmdh) {
        this.tcmdh = tcmdh;
    }

    public String getTcymc() {
        return tcymc;
    }

    public void setTcymc(String tcymc) {
        this.tcymc = tcymc;
    }

    public String getTcys() {
        return tcys;
    }

    public void setTcys(String tcys) {
        this.tcys = tcys;
    }

    public String getTcmsd() {
        return tcmsd;
    }

    public void setTcmsd(String tcmsd) {
        this.tcmsd = tcmsd;
    }

    public String getTcsid() {
        return tcsid;
    }

    public void setTcsid(String tcsid) {
        this.tcsid = tcsid;
    }

    public String getTcksx() {
        return tcksx;
    }

    public void setTcksx(String tcksx) {
        this.tcksx = tcksx;
    }

    public String getTchyd() {
        return tchyd;
    }

    public void setTchyd(String tchyd) {
        this.tchyd = tchyd;
    }

    public String getTcjyx() {
        return tcjyx;
    }

    public void setTcjyx(String tcjyx) {
        this.tcjyx = tcjyx;
    }

    public String getTcfhcd() {
        return tcfhcd;
    }

    public void setTcfhcd(String tcfhcd) {
        this.tcfhcd = tcfhcd;
    }

    public Double getTcysqx() {
        return tcysqx;
    }

    public void setTcysqx(Double tcysqx) {
        this.tcysqx = tcysqx;
    }

    public Double getTcysqj() {
        return tcysqj;
    }

    public void setTcysqj(Double tcysqj) {
        this.tcysqj = tcysqj;
    }

    public String getTckwcf() {
        return tckwcf;
    }

    public void setTckwcf(String tckwcf) {
        this.tckwcf = tckwcf;
    }

    public String getTcjggz() {
        return tcjggz;
    }

    public void setTcjggz(String tcjggz) {
        this.tcjggz = tcjggz;
    }

    public String getTcbhw() {
        return tcbhw;
    }

    public void setTcbhw(String tcbhw) {
        this.tcbhw = tcbhw;
    }

    public String getTcqw() {
        return tcqw;
    }

    public void setTcqw(String tcqw) {
        this.tcqw = tcqw;
    }

    public String getTcms() {
        return tcms;
    }

    public void setTcms(String tcms) {
        this.tcms = tcms;
    }

    public String getTcztx() {
        return tcztx;
    }

    public void setTcztx(String tcztx) {
        this.tcztx = tcztx;
    }

    public String getTcjycd() {
        return tcjycd;
    }

    public void setTcjycd(String tcjycd) {
        this.tcjycd = tcjycd;
    }

    public String getTcpl() {
        return tcpl;
    }

    public void setTcpl(String tcpl) {
        this.tcpl = tcpl;
    }

    public String getTcjlfy() {
        return tcjlfy;
    }

    public void setTcjlfy(String tcjlfy) {
        this.tcjlfy = tcjlfy;
    }

    public String getTcdcms() {
        return tcdcms;
    }

    public void setTcdcms(String tcdcms) {
        this.tcdcms = tcdcms;
    }

    public Double getTcjljj() {
        return tcjljj;
    }

    public void setTcjljj(Double tcjljj) {
        this.tcjljj = tcjljj;
    }

    public Double getTcpxjd() {
        return tcpxjd;
    }

    public void setTcpxjd(Double tcpxjd) {
        this.tcpxjd = tcpxjd;
    }

    public String getTcqttz() {
        return tcqttz;
    }

    public void setTcqttz(String tcqttz) {
        this.tcqttz = tcqttz;
    }

    public Double getTczrzd() {
        return tczrzd;
    }

    public void setTczrzd(Double tczrzd) {
        this.tczrzd = tczrzd;
    }

    public Double getTcnjl() {
        return tcnjl;
    }

    public void setTcnjl(Double tcnjl) {
        this.tcnjl = tcnjl;
    }

    public Double getTcnmcj() {
        return tcnmcj;
    }

    public void setTcnmcj(Double tcnmcj) {
        this.tcnmcj = tcnmcj;
    }

    public Double getTcczl() {
        return tcczl;
    }

    public void setTcczl(Double tcczl) {
        this.tcczl = tcczl;
    }

    public Double getTcysml() {
        return tcysml;
    }

    public void setTcysml(Double tcysml) {
        this.tcysml = tcysml;
    }

    public Double getTcyzzCzltzz() {
        return tcyzzCzltzz;
    }

    public void setTcyzzCzltzz(Double tcyzzCzltzz) {
        this.tcyzzCzltzz = tcyzzCzltzz;
    }

    public Double getTcyzzDzltzz() {
        return tcyzzDzltzz;
    }

    public void setTcyzzDzltzz(Double tcyzzDzltzz) {
        this.tcyzzDzltzz = tcyzzDzltzz;
    }

    public Double getTcyzzCzlbzz() {
        return tcyzzCzlbzz;
    }

    public void setTcyzzCzlbzz(Double tcyzzCzlbzz) {
        this.tcyzzCzlbzz = tcyzzCzlbzz;
    }

    public Double getTcyzzDzlbzz() {
        return tcyzzDzlbzz;
    }

    public void setTcyzzDzlbzz(Double tcyzzDzlbzz) {
        this.tcyzzDzlbzz = tcyzzDzlbzz;
    }

    public Double getTcckzCzltzz() {
        return tcckzCzltzz;
    }

    public void setTcckzCzltzz(Double tcckzCzltzz) {
        this.tcckzCzltzz = tcckzCzltzz;
    }

    public Double getTcckzDzltzz() {
        return tcckzDzltzz;
    }

    public void setTcckzDzltzz(Double tcckzDzltzz) {
        this.tcckzDzltzz = tcckzDzltzz;
    }

    public Double getTcckzCzlbzz() {
        return tcckzCzlbzz;
    }

    public void setTcckzCzlbzz(Double tcckzCzlbzz) {
        this.tcckzCzlbzz = tcckzCzlbzz;
    }

    public Double getTcckzDzlbzz() {
        return tcckzDzlbzz;
    }

    public void setTcckzDzlbzz(Double tcckzDzlbzz) {
        this.tcckzDzlbzz = tcckzDzlbzz;
    }

    public String getTcuserdefine1() {
        return tcuserdefine1;
    }

    public void setTcuserdefine1(String tcuserdefine1) {
        this.tcuserdefine1 = tcuserdefine1;
    }

    public String getTcuserdefine2() {
        return tcuserdefine2;
    }

    public void setTcuserdefine2(String tcuserdefine2) {
        this.tcuserdefine2 = tcuserdefine2;
    }

    public String getTcuserdefine3() {
        return tcuserdefine3;
    }

    public void setTcuserdefine3(String tcuserdefine3) {
        this.tcuserdefine3 = tcuserdefine3;
    }

    @Override
    public String toString() {
        return "Standard{" +
        "id=" + id +
        ", itemId=" + itemId +
        ", holeCode=" + holeCode +
        ", depth=" + depth +
        ", tcxh=" + tcxh +
        ", mainCode=" + mainCode +
        ", secondaryCode=" + secondaryCode +
        ", thirdCode=" + thirdCode +
        ", tcdzsd=" + tcdzsd +
        ", tcdzcy=" + tcdzcy +
        ", tcdchd=" + tcdchd +
        ", tchd=" + tchd +
        ", type=" + type +
        ", name=" + name +
        ", tcmdh=" + tcmdh +
        ", tcymc=" + tcymc +
        ", tcys=" + tcys +
        ", tcmsd=" + tcmsd +
        ", tcsid=" + tcsid +
        ", tcksx=" + tcksx +
        ", tchyd=" + tchyd +
        ", tcjyx=" + tcjyx +
        ", tcfhcd=" + tcfhcd +
        ", tcysqx=" + tcysqx +
        ", tcysqj=" + tcysqj +
        ", tckwcf=" + tckwcf +
        ", tcjggz=" + tcjggz +
        ", tcbhw=" + tcbhw +
        ", tcqw=" + tcqw +
        ", tcms=" + tcms +
        ", tcztx=" + tcztx +
        ", tcjycd=" + tcjycd +
        ", tcpl=" + tcpl +
        ", tcjlfy=" + tcjlfy +
        ", tcdcms=" + tcdcms +
        ", tcjljj=" + tcjljj +
        ", tcpxjd=" + tcpxjd +
        ", tcqttz=" + tcqttz +
        ", tczrzd=" + tczrzd +
        ", tcnjl=" + tcnjl +
        ", tcnmcj=" + tcnmcj +
        ", tcczl=" + tcczl +
        ", tcysml=" + tcysml +
        ", tcyzzCzltzz=" + tcyzzCzltzz +
        ", tcyzzDzltzz=" + tcyzzDzltzz +
        ", tcyzzCzlbzz=" + tcyzzCzlbzz +
        ", tcyzzDzlbzz=" + tcyzzDzlbzz +
        ", tcckzCzltzz=" + tcckzCzltzz +
        ", tcckzDzltzz=" + tcckzDzltzz +
        ", tcckzCzlbzz=" + tcckzCzlbzz +
        ", tcckzDzlbzz=" + tcckzDzlbzz +
        ", tcuserdefine1=" + tcuserdefine1 +
        ", tcuserdefine2=" + tcuserdefine2 +
        ", tcuserdefine3=" + tcuserdefine3 +
        "}";
    }
}
